src/kademlia/dht_storage.cpp

```diff
@@ -268,26 +267,12 @@ namespace
 			torrent_entry* v;
 			if (ti == m_map.end())
 			{
-				// we don't have this torrent, add it
-				// do we need to remove another one first?
-				if (!m_map.empty() && int(m_map.size()) >= m_settings.max_torrents)
+				if (int(m_map.size()) >= m_settings.max_torrents)
```

Comment:
I'm not sure. iirc, I did that just for simplicity. There is a certain cost just to have the torrent entry, so they should probably be capped to something, but the primary limit should probably be the total number of peers